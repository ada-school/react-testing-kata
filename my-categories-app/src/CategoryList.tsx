import React from "react";

interface CategoryListProps {
  categories: ReadonlyArray<any>;
}

export const CategoryList: React.FC<CategoryListProps> = () => {
  return <div style={{ color: "red" }} />;
};
