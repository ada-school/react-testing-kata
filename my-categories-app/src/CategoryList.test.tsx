import { render, screen } from "@testing-library/react";
import React from "react";
import { CategoryList } from "./CategoryList";

test("Se renderiza una lista de categorias", () => {
  const categories = [
    {
      id: 1,
      name: "Categoria 1",
    },
    {
      id: 2,
      name: "Categoria 2",
    },
    {
      id: 3,
      name: "Categoria 3",
    },
  ];
  const renderResult = render(<CategoryList categories={categories} />);
  expect(renderResult.container).toMatchSnapshot();

  categories.forEach((category) =>
    expect(screen.getByText(category.name).tagName).toEqual("li")
  );

  /*
<ul>
    <li>sd</li
    <li>sd</li
</ul>
*/
});
