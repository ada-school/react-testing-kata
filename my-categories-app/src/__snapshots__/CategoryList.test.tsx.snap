// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se renderiza una lista de categorias 1`] = `
<div>
  <div
    style="color: red;"
  />
</div>
`;
